toolchain/driver/driver.cpp

```diff
@@ -671,7 +671,17 @@ auto Driver::Compile(const CompileOptions& options) -> RunResult {
   // Prepare CompilationUnits before building scope exit handlers.
   StreamDiagnosticConsumer stream_consumer(error_stream_);
   llvm::SmallVector<std::unique_ptr<CompilationUnit>> units;
-  units.reserve(options.input_filenames.size());
+
+  // Directly insert the core package into the compilation units.
+  // TODO: Should expand this into a more rich system to search for the core
+  // package source code.
+  if (options.prelude_import) {
+    units.push_back(std::make_unique<CompilationUnit>(
+        this, options, &stream_consumer, "core/prelude.carbon"));
+  }
+
+  // Add the input source files.
+  units.reserve(units.size() + options.input_filenames.size());
```

Comment:
```suggestion
    units.reserve(options.input_filenames.size() + 1);
    units.push_back(std::make_unique<CompilationUnit>(
        this, options, &stream_consumer, "core/prelude.carbon"));
  } else {
    units.reserve(options.input_filenames.size());
  }

  // Add the input source files.
```

Thoughts on switching to reserve prior to pushing?

Reply:
Sure, used a slightly different code pattern. PTAL.

Reply:
We'll probably want to switch (we will swiftly have more than one prelude file) but this is fine for now.